includes/Services/FlowService.php

```diff
@@ -46,25 +91,23 @@ public static function update_flow_data( $params ) {
 			);
 		}
 
-		$flow_data = self::get_flow_data();
+		$default_data = self::get_default_data();
+		$mismatch_key = self::find_mismatch_key( $params, $default_data );
```

Comment:
since we do not iterate through `$params` in the recursive call, a check is added using the `array_diff` of the keys at the level in the array and an error is thrown accordingly. Have removed the mismatch_key logic, thanks.

Reply:
Do we need to explicitly check `$params` was my initial question, seems like irrespective of what the keys are in `$params` we take the keys from `$default_data` and take the value from `$params[$key]` after an `isset` check?

Reply:
The explicit check is required if we need an error to be thrown gracefully for the wrong key entered by the user.
Otherwise, without that, there would not be an error and the keys would be retained as per the blueprint.